eng/tox/sanitize_setuppy.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# This script can process and update setup.py for e.g. update required version to dev
+
+import argparse
+import sys
+import os
+import logging
+import glob
+from packaging.specifiers import SpecifierSet
+from pkg_resources import Requirement
+from pypi_tools.pypi import PyPIClient
+
+root_dir = os.path.abspath(os.path.join(os.path.abspath(__file__), "..", "..", ".."))
+setup_parser_path = os.path.abspath(os.path.join(root_dir, "eng", "versioning"))
+
+sys.path.append(setup_parser_path)
+from setup_parser import get_install_requires, parse_setup
+
+
+def update_requires(setup_py_path, requires_dict):
+    # This method changes package requirement by overriding the specifier
+    contents = []
+    with open(setup_py_path, "r") as setup_file:
+        contents = setup_file.readlines()
+
+    # find and replace all existing package requirement with new requirement
+    for i in range(0, len(contents) - 1):
+        keys = [k for k in requires_dict.keys() if k in contents[i]]
+        for key in keys:
+            contents[i] = contents[i].replace(key, requires_dict[key])
+
+    with open(setup_py_path, "w") as setup_file:
+        setup_file.writelines(contents)
+
+
+def is_required_version_on_pypi(package_name, spec):
```

Comment:
The way you use this, you can just `return versions`. A non empty array is `truthy` and an empty array is `falsy`